### PR TITLE
Add skiplist collection stage

### PR DESCRIPTION
**Pull request by 1ucian0**
Some tests in some platforms take really long time.

However, timing out tests is hard. They need to be in their own killable process and so on. This might affect their benchmark numbers. This solution implements a collect-skip-list stage, in which tests that take too long are collected. Then, when running, these tests are skipped.

This PR does several things:
 * Adds an explanation of usage in the README.md file
 * Changes the `pytest-benchmark` dependency for a modified version that supports skiplisting
 * Extended the header report with the parameters used in `--timeout-skip-list`
 * Adds an initial `skiplist.txt`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +28,7 @@ To run the benchmarks in the default configuration from inside the environment i
 ```bash
 python -m pytest --benchmark-min-rounds=1 benchpress/*_gym
 ```
-where `*` is one of the frameworks that you want to test, and which matches the environment you are in.  Here `--benchmark-min-rounds=1`sets the minimum number of repeated trials to 1, which will save a great deal of time
+where `*` is one of the frameworks that you want to test, and which matches the environment you are in.  Here `--benchmark-min-rounds=1` sets the minimum number of repeated trials to 1, which will save a great deal of time
```
Comment (1ucian0) [inline README.md:31]: unrelated typo

**Comment by nonhermitian** [comment]
I finally got around to testing this, but on a fresh conda env for tket I get this when running the tests:

self = <pytest_benchmark.fixture.BenchmarkFixture object at 0x158d66bd0>
function_to_benchmark = <function TestWorkoutCircuitManipulate.test_QV100_basis_change.<locals>.result at 0x14a34e980>
args = (), kwargs = {}

    def _check_timeout(self, function_to_benchmark, args, kwargs):
>       from wrapt_timeout_decorator import timeout, set_subprocess_starting_method
E       ModuleNotFoundError: No module named 'wrapt_timeout_decorator'

../../mambaforge/envs/tket-test/lib/python3.12/site-packages/pytest_benchmark/fixture.py:168: ModuleNotFoundError

**Comment by 1ucian0** [comment]
oh. I forgot to extend the reqs. Done in 2b17c94f9f3261ee6e21f26364a8722555e8197b

**Comment by nonhermitian** [comment]
Ok this is super cool.  Does exactly what we needed it to do!

**Review by nonhermitian — APPROVED** [review]

**Pull request merged**